### ISSUE TITLE
Minimum training data warning for test set during evaluation is confusing

### ISSUE THREAD

**Issue by TatianaParshina** [RasaHQ/rasa_nlu#1615]
**Rasa NLU version**: 0.13.8

**Operating system** (windows, osx, ...): Windows

**Content of model configuration file**:
```yml
language: "en"

pipeline:
- name: "tokenizer_whitespace"
- name: "ner_crf"
- name: "intent_featurizer_count_vectors"
- name: "intent_classifier_tensorflow_embedding"
intent_tokenization_flag: true
intent_split_symbol: "_"
```

**Issue**:
I am trying to evaluate model using test data set:

`python -m rasa_nlu.evaluate --data Test_dataset.json --model nlu_model`

Test set has around 20% of samples compared with a train set. For example, if the train set has 6 samples per intent, then the train set will have only 1.

In this case, I got a warning:

> \rasa_nlu\training_data\training_data.py:1 91: UserWarning: Intent 'TestIntent' has only 1 training examples!
> Minimum is 2, training may fail.

Why is the test set trained for evaluation purpose if the training has already been performed on the training dataset?

The issue has created after proposal from [forum](https://forum.rasa.com/t/why-is-the-test-set-trained-for-evaluation-purpose/3981)
